glean-core/src/database/mod.rs

```diff
@@ -0,0 +1,205 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use rkv::{Rkv, SingleStore, StoreOptions};
+use std::collections::btree_map::Entry;
+use std::collections::BTreeMap;
+use std::fs;
+
+use crate::metrics::Metric;
+use crate::Lifetime;
+
+#[derive(Debug)]
+pub struct Database {
+    rkv: Option<Rkv>,
+    // Metrics with 'application' lifetime only live as long
+    // as the application lives: they don't need to be persisted
+    // to disk using rkv. Store them in a map.
+    app_lifetime_data: BTreeMap<String, Metric>,
+}
+
+impl Database {
+    pub fn new() -> Self {
+        Self {
+            rkv: None,
```

Comment:
We can probably later move away from option here and instantiate on `new` and then handle the init part on the glean object only